newIDE/app/src/EventsFunctionsExtensionsLoader/index.js

```diff
@@ -380,9 +385,14 @@ function generateBehavior(
 
       if (eventsFunction.isPrivate()) instructionOrExpression.setPrivate();
 
-      instructionOrExpression
-        .setIncludeFile(includeFile)
-        .setFunctionName(eventsFunctionMangledName);
+      instructionOrExpression.setIncludeFile(includeFile);
+
+      if (
+        eventsFunction.isAsync() &&
+        !!instructionOrExpression.setAsyncFunctionName
```

Comment:
In typescript, this would be enough to narrow the type down to `gdInstructionMetadata`... Any pointer on how to do this with flow?